moveit_ros/visualization/rviz_plugin_render_tools/src/robot_state_visualization.cpp

```diff
@@ -135,7 +135,8 @@ void RobotStateVisualization::updateHelper(const moveit::core::RobotStateConstPt
       continue;
     }
     rviz::Color rcolor(color.r, color.g, color.b);
-    const EigenSTL::vector_Isometry3d& ab_t = attached_body->getFixedTransforms();
+    const EigenSTL::vector_Isometry3d& ab_t =
+        attached_body->getShapePosesInLinkFrame();  // TODO(felixvd): Check if this can be done via getShapePoses()
```

Comment:
No. Shapes are visualized a part of the link's scene-graph node. Thus poses relative to the link are needed.